state/address.go

```diff
@@ -275,13 +275,21 @@ func (st *State) apiHostPortsForCAAS(public bool) (addresses []network.SpaceHost
 		logger.Debugf("getting api hostports for CAAS: public %t, addresses %v", public, addresses)
 	}()
 
-	controllerConfig, err := st.ControllerConfig()
+	// We are fetching info about the controller cloud service,
+	// so need the controller state.
+	ctrlSt, err := st.newStateNoWorkers(st.ControllerModelUUID())
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	defer func() { _ = ctrlSt.Close() }()
+
+	controllerConfig, err := ctrlSt.ControllerConfig()
```

Comment:
You don't need a controller state to read controller config, you can read that from any state instance.

Reply:
we use the ctrlSt below to get the controller cloud service info, so it's convenient to use it here also